pkg/controller/controller.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	framework "github.com/GoogleCloudPlatform/kubernetes/pkg/controller/framework"
+)
+
+// Controller represents a long-lived process that watches objects in the system and can respond to events.  Plugins
+// implement factory methods for creating framework.Controller instances, such as those created by framework.NewInformer()
+// or framework.New(config).
+type Controller interface {
```

Comment:
I misunderstood what you were trying to do a little bit-- I thought you were using this type to represent a list of plugins, not a list of controllers that compose a single plugin.

So now that I understand what you were trying to do, my comment is: don't help people write their controller, because a list of framework.Controllers isn't a good model for that. Instead, define an interface for the Run() method or whatever, so that people can implement it and have their controller be a plugin in your plugin system.